src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/CAS1SubjectAccessRequestRepository.kt

```diff
@@ -0,0 +1,589 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity
+
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Repository
+import java.time.LocalDateTime
+
+@Repository
+class CAS1SubjectAccessRequestRepository(
+  jdbcTemplate: NamedParameterJdbcTemplate,
+) : SubjectAccessRequestRepositoryBase(jdbcTemplate) {
+
+  fun getApprovedPremisesApplicationsJson(
+    crn: String?,
+    nomsNumber: String?,
+    startDate: LocalDateTime?,
+    endDate: LocalDateTime?,
+  ): String {
+    val result = jdbcTemplate.queryForMap(
+      """
+select 
+	json_agg(app) as json
+from
+	(
+	select
+		a.id,
+		apa.name,
+		a.crn,
+		a.noms_number,
+		a."data",
+		a."document",
+		a.created_at,
+		a.submitted_at,
+		created_by_user.name as created_by_user,
+		app_user."name" as application_user_name ,
+		apa.event_number,
+		apa.is_womens_application,
+		apa.offence_id,
+		apa.conviction_id,
+		apa.risk_ratings,
+		apa.release_type,
+		apa.arrival_date,
+		apa.is_withdrawn,
+		apa.withdrawal_reason,
+		apa.other_withdrawal_reason ,
+		apa.is_emergency_application,
+		apa.target_location ,
+		apa.status,
+		apa.inmate_in_out_status_on_submission,
+		apa.sentence_type,
+		apa.notice_type,
+		apa.ap_type,
+		case_manager."name" as case_manager_name,
+		apa.case_manager_is_not_applicant
+	from
+		approved_premises_applications apa
+	join 
+        applications a on
+		a.id = apa.id
+	left join 
+        cas_1_application_user_details case_manager on
+		    case_manager.id = apa.case_manager_cas1_application_user_details_id
+	left join 
+        cas_1_application_user_details app_user on
+		    app_user.id = apa.applicant_cas1_application_user_details_id
+	left join 
+        users created_by_user on
+		    created_by_user.id = a.created_by_user_id
+	where
+		(a.crn = :crn
+			or a.noms_number = :noms_number )
+		and (:start_date is null
+			or a.created_at >= :start_date)
+		and (:end_date is null
+			or a.created_at <= :end_date) 
+) app;
+      """.trimIndent(),
+      MapSqlParameterSource().addSarParameters(
+        crn,
+        nomsNumber,
+        startDate,
+        endDate,
+      ),
+    )
+    return toJsonString(result)
+  }
+  fun getApprovedPremisesApplicationTimeLineJson(
+    crn: String?,
+    nomsNumber: String?,
+    startDate: LocalDateTime?,
+    endDate: LocalDateTime?,
+  ): String {
+    val result = jdbcTemplate.queryForMap(
+      """
+  
+  select 
+  	json_agg(apptimeline) as json
+  from(
+      select
+          a.id as application_id,
+          a.service,
+          a.crn,
+          a.noms_number,
+          atn.body,
+          atn.created_at,
+          u."name" as user_name
+      from
+      application_timeline_notes atn
+      inner join users u on
+          u.id = atn.created_by_user_id
+      inner join applications a on
+          atn.application_id = a.id
```

Comment:
i'm not proposing we change this as it'll be quite a bit of work, but if we add any additional tables it may be worth considering passing in a list of application ids for the given CRN/NOMS as I think we're repeating the logic to find these applications in most queries.

Reply:
yeah i'd considered retrieving the application and assessment id's first then passing them in as an ```in``` clause (or PSQL equivalent), but thought it was optimising early. That said this format allows us to pull the queries out and run these manually perhaps? open to discussing this though.

Reply:
will leave this till a later iteration for us to think about optimising.